### PR TITLE
[4 of 4] New Signed Out Homepage (promoted section)

### PR DESCRIPTION
**Pull request by wgranger**
**Describe your changes:**
This branch includes changes to the promoted and featured sections to be merged into the new-signed-out-homepage branch. The featured project section is also updated along with this branch. 

_Note_: This branch will merge into that other branch and not master.

https://promoted-redesign.pfe-preview.zooniverse.org/

# Review Checklist

- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [x] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch?
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)

## Optional

- [x] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

### CODE REVIEW

File: app/pages/home-common/featured-project.jsx [hunk 1]
```diff
@@ -16,11 +16,13 @@ const FeaturedProject = () => {
     <section className="home-featured">
       <h3 className="secondary-kicker">Featured Project</h3>
       <div className="home-featured-images">
-        <img role="presentation" src="./assets/featured-projects/featured-project-20170407-etchacell.jpg" />
+        <div><img role="presentation" src="./assets/featured-projects/featured-project-20170508-michiganzoomin01.jpg" /></div>
```
Comment (rogerhutchings) [inline app/pages/home-common/featured-project.jsx]: Do you need these `div`s, or could you just add the styles directly to the images?

File: app/pages/home-for-user/index.jsx [hunk 2]
```diff
@@ -306,12 +306,12 @@ export default class HomePageForUser extends React.Component {
 
           </Pullout>
         </div>
-        
-        <div className="home-page-for-user on-home-page-logged-in">
+
+        <div>
```
Comment (rogerhutchings) [inline app/pages/home-for-user/index.jsx]: Again with these `div`s

File: app/pages/home-not-logged-in/promoted.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        })}
+      </div>
+
+      <Link to="/projects" className="primary-button primary-button--light">See All Projects</Link>
+    </section>
+  );
+};
+
+HomePagePromoted.propTypes = {
+  promotedProjects: React.PropTypes.arrayOf(React.PropTypes.object)
```
Comment (rogerhutchings) [inline app/pages/home-not-logged-in/promoted.jsx]: I'm pretty sure the linter kicks off about using `object`, you can use `shape()` and pass in an object of required properties instead

File: app/pages/home-not-logged-in.jsx [hunk 4]
```diff
@@ -78,6 +82,21 @@ export default class HomePage extends React.Component {
     });
   }
 
+  getPromotedProjects() {
+    apiClient.type('projects').get({ id: Object.keys(FEATURED_PROJECTS), cards: true })
```
Comment (rogerhutchings) [inline app/pages/home-not-logged-in.jsx]: is there any kind of loading state needed here?

File: app/pages/home-common/featured-project.jsx (hunk 1 above)
Comment (wgranger) [inline app/pages/home-common/featured-project.jsx]: Flexbox wasn't cooperating with plain `img` tags, but wrapping them in a `div` made them responsive.

File: app/pages/home-for-user/index.jsx (hunk 2 above)
Comment (wgranger) [inline app/pages/home-for-user/index.jsx]: These can likely be removed.

File: app/pages/home-not-logged-in/promoted.jsx (hunk 3 above)
Comment (wgranger) [inline app/pages/home-not-logged-in/promoted.jsx]: The linter doesn't mind if the objects are specified within an array, but I can always shape the object if we need that extra specificity.

File: app/pages/home-not-logged-in.jsx (hunk 4 above)
Comment (wgranger) [inline app/pages/home-not-logged-in.jsx]: It's a pretty small request, but I can put a spinner in there to be safe.

**Comment by rogerhutchings** [comment]
Cool. It's approved, I won't merge so you can carry out your own merge strategy @wgranger

**Comment by wgranger** [comment]
Ha. Perfect. I think it's best to get them all approved first, then merge individually to check the main branch (`new-signed-out-homepage`) looks good, then merge into master.

File: app/pages/home-not-logged-in.jsx [hunk 5]
```diff
@@ -89,6 +92,21 @@ export default class HomePage extends React.Component {
     );
   }
 
+  getPromotedProjects() {
+    apiClient.type('projects').get({ id: Object.keys(FEATURED_PROJECTS), cards: true })
+    .then((projects) => {
+      Promise.all(projects.map((project) => {
```
Comment (rogerhutchings) [inline app/pages/home-not-logged-in.jsx]: I don't think this needs to be a `Promise.all`, you can just return `projects.map()...` as there's nothing async in here

Comment (rogerhutchings) [inline app/pages/home-not-logged-in.jsx]: Oh - also, there's no catch, so just log an error or something just in case.

**Comment by wgranger** [comment]
@rogerhutchings Fixed the recent suggestion with the most recent commit. Also addressed a warning with nested anchor tags from the `FlexibleLink` component.

**Comment by rogerhutchings** [comment]
LGTM!

**Pull request merged**